--- kvirc-4.2.0/src/kvilib/ext/avdevice/videodevice.cpp	2012-01-13 15:53:32.000000000 +0200
+++ kvirc-4.2.0/src/kvilib/ext/avdevice/videodevice.cpp	2013-04-24 01:41:48.262331119 +0300
@@ -3089,8 +3089,12 @@
 		case V4L2_CID_GAIN:			return __tr("Gain");
 		case V4L2_CID_HFLIP:			return __tr("Horizontal Flip");
 		case V4L2_CID_VFLIP:			return __tr("Vertical Flip");
+#ifdef V4L2_CID_HCENTER			// since kernel-3.9.0
 		case V4L2_CID_HCENTER:			return __tr("Horizontal Center");
+#endif
+#ifdef V4L2_CID_VCENTER			// since kernel-3.9.0
 		case V4L2_CID_VCENTER:			return __tr("Vertical Center");
+#endif
 #ifdef V4L2_CID_POWER_LINE_FREQUENCY	// since kernel 2.6.25
 		case V4L2_CID_POWER_LINE_FREQUENCY:	return __tr("Power Line Frequency");
 		case V4L2_CID_HUE_AUTO:			return __tr("Automatic Hue");
