diff -up KVIrc-master/src/kvirc/ui/KviInputHistory.h.omv~ KVIrc-master/src/kvirc/ui/KviInputHistory.h
--- KVIrc-master/src/kvirc/ui/KviInputHistory.h.omv~	2022-08-10 20:35:05.866465713 +0200
+++ KVIrc-master/src/kvirc/ui/KviInputHistory.h	2022-08-10 20:35:13.190278079 +0200
@@ -39,10 +39,9 @@
 
 #include "kvi_settings.h"
 
+#include <QString>
 #include <vector>
 
-class QString;
-
 #define KVI_INPUT_MAX_GLOBAL_HISTORY_ENTRIES 500
 #define KVI_INPUT_MAX_LOCAL_HISTORY_ENTRIES 50
 
